encord/objects/ontology_labels_impl.py

```diff
@@ -1056,26 +1077,99 @@ def initialise_labels(
             overwrite: If the label row was already initialised, you need to set this flag to `True` to overwrite the
                 current labels with the labels stored in the Encord server. If this is `False` and the label row was
                 already initialised, this function will throw an error.
+            bundle: If not passed, initialisation is performed independently. If passed, it will be delayed and
+                initialised along with other objects in the same batch once batch is executed
         """
         if self.is_labelling_initialised and not overwrite:
             raise LabelRowError(
                 "You are trying to re-initialise a label row that has already been initialised. This would overwrite "
                 "current labels. If this is your intend, set the `overwrite` flag to `True`."
             )
 
-        get_signed_url = False
-        if self.label_hash is None:
-            label_row_dict = self._project_client.create_label_row(self.data_hash)
-        else:
-            label_row_dict = self._project_client.get_label_row(
+        if bundle is not None:
+            self.__batched_initialise(
```

Comment:
Promises/future approach just ended up being overcomplicated, so I decided it would be overkill for the current situation. So opted in for the simpler approach until we really need something more powerful